### PR TITLE
Added option to accept multiple erc20 tokens

### PR DESCRIPTION
**Pull request by chironjit**
Changes to the code:
1) Added mapping to store addresses of accepted ERC20
2) Added erc20 selector to claimMembershipNft function
3) Added erc20 interface subset (mainly to add decimals() to interface)
4) Changed functions that allow the addition on new erc20 additions + viewing already accepted erc20 tokens
5) Moved up `whitelist[member].hasMinted = true;` before minting to prevent possible edge case reentrancy vuln

### CODE REVIEW

**Comment by Totenfluch** [comment]
doesn't the mapping start at 0?

File: contracts/MembershipNft.sol [hunk 1]
```diff
@@ -31,7 +45,7 @@ contract PretzelDAO_Membership is IERC721Metadata, ERC721Enumerable, Ownable {
     ) ERC721("PretzelDAO Membership", "MPRTZL") {
         membershipPriceInToken = _membershipPriceInToken;
         membershipYear = _membershipYear;
-        membershipPriceTokenAddress = _membershipPriceTokenAddress;
+        membershipPriceTokenAddress[1] = _membershipPriceTokenAddress;
```
Comment (Totenfluch) [inline contracts/MembershipNft.sol:48]: should be 0?

Comment (chironjit) [reply contracts/MembershipNft.sol:48]: Yes, it should be 0. My bad

**Comment by chironjit** [comment]
Closing pull request to fix array numbering oversight

**Pull request closed without merging**